### PR TITLE
Unify reg smb/v1

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3315

### CODE REVIEW

File: src/app-layer-smb.c [hunk 1]
```diff
@@ -230,95 +38,11 @@ static SuricataFileContext sfc = { &sbcfg };
...
-
-    /** SMB */
-    if (AppLayerProtoDetectConfProtoDetectionEnabled("tcp", proto_name)) {
-        AppLayerProtoDetectRegisterProtocol(ALPROTO_SMB, proto_name);
-        if (SMBRegisterPatternsForProtocolDetection() < 0)
-            return;
-
-        rs_smb_init(&sfc);
-
-        if (RunmodeIsUnittests()) {
```
Comment (inashivb) [inline src/app-layer-smb.c:247]: If there are no C tests is this conditional still required? If SMB is used by other protos, is that when we need this?
This has not been translated to Rust

File: src/app-layer-smb.c [hunk 2]
```diff
@@ -230,95 +38,11 @@ static SuricataFileContext sfc = { &sbcfg };
...
-        ConfNode *p = ConfGetNode("app-layer.protocols.smb.stream-depth");
-        if (p != NULL) {
-            uint32_t value;
-            if (ParseSizeStringU32(p->val, &value) < 0) {
-                SCLogError(SC_ERR_SMB_CONFIG, "invalid value for stream-depth %s", p->val);
-            } else {
-                stream_depth = value;
-            }
-        }
-        SCLogConfig("SMB stream depth: %u", stream_depth);
```
Comment (inashivb) [inline src/app-layer-smb.c:315]: Although there is just TCP registration here, to be sure, is that all we need?

File: rust/src/smb/smb.rs [hunk 3]
```diff
@@ -52,6 +52,12 @@ use crate::smb::events::*;
 use crate::smb::files::*;
 use crate::smb::smb2_ioctl::*;
 
+pub const MIN_REC_SIZE: u16 = 32+4; // SMB hdr + nbss hdr
+pub const FLOW_DIR_REVERSED: u32 = BIT_U32!(26);
```
Comment (victorjulien) [inline rust/src/smb/smb.rs:57]: this should probably move somewhere generic as its a flag from the flow engine

File: rust/src/smb/smb.rs [hunk 4]
```diff
@@ -2007,11 +2033,15 @@ pub extern "C" fn rs_smb_state_get_tx(state: &mut SMBState,
 // for use with the C API call StateGetTxIterator
 #[no_mangle]
 pub extern "C" fn rs_smb_state_get_tx_iterator(
-                                      state: &mut SMBState,
-                                      min_tx_id: u64,
-                                      istate: &mut u64)
-                                      -> applayer::AppLayerGetTxIterTuple
+_ipproto: u8,
```
Comment (victorjulien) [inline rust/src/smb/smb.rs:2037]: formatting looks a bit off

File: src/app-layer-smb.c (hunk 1 above)
Comment (victorjulien) [reply src/app-layer-smb.c:247]: I don't see a reason to keep it if there are no tests.

File: src/app-layer-smb.c (hunk 2 above)
Comment (victorjulien) [reply src/app-layer-smb.c:315]: Yes we support smb over tcp only.

**Comment by victorjulien** [comment]
Minor issues but looks good otherwise. Passes my QA as well.

**Comment by inashivb** [comment]
Replaced with #6209

**Pull request closed without merging**